xmodule/library_sourced_block.py

```diff
@@ -132,28 +245,96 @@ def validate(self):
         validation = super().validate()
         validation = StudioValidation.copy(validation)
 
-        if not self.source_block_ids:
+        if not self.source_library_id:
             validation.set_summary(
                 StudioValidationMessage(
                     StudioValidationMessage.NOT_CONFIGURED,
-                    _("No XBlock has been configured for this component. Use the editor to select the target blocks."),
+                    _("A library has not been selected yet."),
                     action_class='edit-button',
-                    action_label=_("Open Editor")
+                    action_label=_("Select a Library.")
                 )
             )
+            return validation
+
+        self._validate_library_version(validation, self.tools, self.source_library_version, self.source_library_key)
         return validation
 
-    @XBlock.handler
-    def submit_studio_edits(self, data, suffix=''):
+    def source_library_values(self):
         """
-        Save changes to this block, applying edits made in Studio.
+        Return a list of possible values for self.source_library_id
+        """
+        user_perms = self.runtime.service(self, 'studio_user_permissions')
+        all_libraries = [
+            (key, bleach.clean(name)) for key, name in self.tools.list_available_libraries()
+            if user_perms.can_read(key) or self.source_library_id == str(key)
+        ]
+        all_libraries.sort(key=lambda entry: entry[1])  # Sort by name
+        if self.source_library_id and self.source_library_key not in [entry[0] for entry in all_libraries]:
+            all_libraries.append((self.source_library_id, _("Invalid Library")))
+        all_libraries = [("", _("No Library Selected"))] + all_libraries
+        values = [{"display_name": name, "value": str(key)} for key, name in all_libraries]
+        return values
+
+    @property
+    def non_editable_metadata_fields(self):
+        non_editable_fields = super().non_editable_metadata_fields
+        non_editable_fields.extend([
+            LibrarySourcedBlock.source_block_ids,
+            LibrarySourcedBlock.source_library_version,
+        ])
+        return non_editable_fields
+
+    @property
+    def source_library_key(self):
```

Comment:
we're using this method in both random and sourced blocks. Looks like it will be better to move this method in the `library_tools` module (to follow the DRY principle)